src/error.rs

```diff
@@ -30,20 +30,25 @@ use data::Data;
 #[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Debug, RustcEncodable, RustcDecodable)]
 /// represents response errors
 pub enum ResponseError {
-    /// data not found
-    NoData,
+    /// Abort is for user to indicate that the state can be dropped;
+    /// if received by routing, it will drop the state.
+    Abort,
     /// invalid request
-    InvalidRequest,
-    /// failure to store data
-    FailedToStoreData(Data)
+    InvalidRequest(Data),
+    /// failure to complete request for data
+    FailedRequestForData(Data),
+    /// had to clear Sacrificial Data in order to complete request
+    HadToClearSacrificial(usize),
 }
 
 impl error::Error for ResponseError {
     fn description(&self) -> &str {
         match *self {
-            ResponseError::NoData => "No Data",
-            ResponseError::InvalidRequest => "Invalid request",
-            ResponseError::FailedToStoreData(_) => "Failed to store data",
+            ResponseError::Abort => "Abort",
+            ResponseError::InvalidRequest(_) => "Invalid request",
+            ResponseError::FailedRequestForData(_) => "Failed request for data",
+            ResponseError::HadToClearSacrificial(size) => "Had to clear {:?} bytes of Sacrificial
```

Comment:
I agree!  I would want to minimise what Routing knows/defines about ExternalRequest ExternalResponse and associated errors; it's stupid that the user can only pass these error types - as one example